index.js

```diff
@@ -31,54 +17,33 @@ module.exports = ({
   serverConfig = {},
   clientConfig = {},
   devConfig = {},
-  port = 8080
+  port = 8080,
+  testOpts,
+  extras
 }) => {
-  if (action === 'lint') {
-    return lint({ inputDir })
+  let config = {}
+
+  if (action === 'build' || action === 'serve') {
+    config = configMerge({
+      sharedConfig,
+      serverConfig,
+      clientConfig,
+      devConfig,
+      inputDir,
+      outputDir,
+      basePath
+    })
   }
 
-  const config = {
-    inputDir,
-    outputDir,
-    basePath
-  }
-
-  const shared = smarterMerge(
-    defaultSharedConfig,
-    sharedConfig
-  )
-  const server = applyConfig(
-    smarterMerge(
-      shared,
-      smarterMerge(
-        defaultServerConfig,
-        serverConfig
-      )
-    ),
-    config
-  )
-  const client = applyConfig(
-    smarterMerge(
-      shared,
-      smarterMerge(
-        defaultClientConfig,
-        clientConfig
-      )
-    ),
-    config
-  )
-  const dev = smarterMerge(
-    defaultDevConfig,
-    devConfig
-  )
-
   return actions[action]({
     inputDir,
     outputDir,
     basePath,
-    serverConfig: server,
-    clientConfig: client,
-    devConfig: dev,
-    port
+    serverConfig: config.server || serverConfig,
```

Comment:
Why default to `serverConfig` ? configMerge _has_ to return a value for this here. Also the action can do default destructuring:

```
// ./lib/serve.js
module.exports = ({ serverConfig = {} }) => { /* Stuffs */ }
```